### ISSUE TITLE
Linestyle chooser

### ISSUE THREAD

**Issue by wico-elfgen** [flameshot-org/flameshot#3666]
### Feature Description

I think it would be a great feature if we could choose a line style for the different tools (i.e. https://doc.qt.io/qt-6/qt.html#PenStyle-enum). In my mind the feature would work like this: When you left click and hold on a Tool selector icon (e.g. line tool) you will see a menu with the different line styles similar to the color menu on right click hold and can then choose which line style you want to use. I would like to implement this feature myselft, but I only want to invest the effort, if there is a chance of it being merged in. So let me know if that is a good feature that you would like to be contributed. Thanks.

**Comment by serebryakov02** [comment]
@borgmanJeremy @mmahmoudian What are your thoughts on this? I like the customization idea, and it is quite similar to something I recently implemented.

**Comment by mmahmoudian** [comment]
@serebryakov02 I believe the same style can be adopted for this. But I have a feeling that sooner or later we will get asked to be able to change the line of the arrow too. I wonder how feasible it is to have the same line style selection for the arrow as well.

**Comment by serebryakov02** [comment]
@mmahmoudian Yes, this should be feasible. The Arrow tool already draws its shaft using `QPen`, so the selected line style can also be applied there while keeping the arrowhead filled, using the same Tool Settings approach as #4769.
I overlooked that @wico-elfgen wants to implement this feature, so I will leave it to him. Happy to help if needed.

**Comment by mmahmoudian** [comment]
@wico-elfgen Sorry that it took over 2 years to get to this feature request. Unfortunately this is the nature of community-driven OpenSource software.

Would you still be interested to implement this?
